### PR TITLE
Seyfert Galaxies Analysis

### PR DESCRIPTION
**Pull request by tomaskontrimas**
Add support for stacking source specific fluxes (mapped analyses).

### CODE REVIEW

**Review by martwo — COMMENTED** [review]
New flux models should be implemented in the flux_model.py file. The flux.py file is deprecated at this point. New models should derive from flux_model.FluxModel.

File: skyllh/core/llhratio.py [hunk 1]
```diff
@@ -1341,32 +1360,46 @@ def __call__(self, fitparam_values):
...
-        # Loop over the detector signal efficiency instances for the first and
-        # only source hypothesis group.
+        Y = []
+        Y_grads = []
 
-        # currently this only works for one source hypothesis group
-        Y, Y_grads = self.detsigyield_arr[0](self._src_arr_list[0], fitparams_arr)
+        # Loop over detector signal efficiency instances for each source
+        # hypothesis group in source hypothesis group manager.
+        for (j, detsigyield) in enumerate(self._detsigyield_arr):
```
Comment (tomaskontrimas) [inline skyllh/core/llhratio.py:1378]: Change index to `g` (group) as `j` is used to index over datasets in the documentation.

File: skyllh/core/source_hypothesis.py [hunk 2]
```diff
@@ -164,3 +165,24 @@ def get_detsigyield_implmethod_list_by_src_idx(self, src_idx):
...
+        -------
+        map_list : list
+            The list that maps hash of the source hypothesis fluxmodel to
+            the corresponding source indices list in that source hypothesis
+            group.
+        """
+        map_list = [
+            # Mapping tuple.
+            (
+                make_params_hash({'fluxmodel': str(shg.fluxmodel)}),
```
Comment (tomaskontrimas) [inline skyllh/core/source_hypothesis.py:182]: Use hash instead of str. Also add __hash__ to fluxmodel

**Comment by tomaskontrimas** [comment]
> New flux models should be implemented in the flux_model.py file. The flux.py file is deprecated at this point. New models should derive from flux_model.FluxModel.

@martwo and I decided to currently leave it as is and update it in the future, when skyllh's core code fully supports `flux_model.FluxModel` structure.

File: skyllh/core/llhratio.py (hunk 1 above)
Comment (tomaskontrimas) [reply skyllh/core/llhratio.py]: Fixed

File: skyllh/core/analysis.py [hunk 3]
```diff
@@ -1471,38 +1471,46 @@ def construct_llhratio(self, minimizer, ppbar=None):
...
+             len(self.dataset_list)),
+            dtype=object
+        )
 
-            detsigyield = detsigyield_implmethod.construct_detsigyield(
-                dataset, data, fluxmodel, data.livetime, ppbar=pbar)
-            detsigyield_list.append(detsigyield)
-            pbar.increment()
-        pbar.finish()
+        for (k, shg) in enumerate(self._src_hypo_group_manager._src_hypo_group_list):
```
[the remark this answers is not in the document]
Comment (tomaskontrimas) [reply skyllh/core/analysis.py]: Fixed

File: skyllh/core/source_hypothesis.py (hunk 2 above)
Comment (tomaskontrimas) [reply skyllh/core/source_hypothesis.py]: Updated

File: skyllh/core/source_hypothesis.py [hunk 4]
```diff
@@ -164,3 +165,24 @@ def get_detsigyield_implmethod_list_by_src_idx(self, src_idx):
         """
         gidx = self._sidx_to_gidx_gsidx_map_arr[src_idx,0]
         return self._src_hypo_group_list[gidx]._detsigyield_implmethod_list
+
+    def get_fluxmodel_to_source_map(self):
+        """Retrieves the list of tuples mapping fluxmodel to the source indices.
+
+        map_list
```
[the remark this answers is not in the document]
Comment (tomaskontrimas) [reply skyllh/core/source_hypothesis.py]: Fixed

File: skyllh/core/source_hypothesis.py [hunk 5]
```diff
@@ -164,3 +165,24 @@ def get_detsigyield_implmethod_list_by_src_idx(self, src_idx):
...
+        -------
+        map_list : list
```
[the remark this answers is not in the document]
Comment (tomaskontrimas) [reply skyllh/core/source_hypothesis.py]: Fixed

File: skyllh/core/dataset.py [hunk 6]
```diff
@@ -980,6 +995,23 @@ def get_binning_definition(self, name):
...
+    def remove_binning_definition(self, name):
+        """Removes the BinningDefinition object from the dataset.
+
+        Parameters
+        ----------
+        name : str
+            The name of the binning definition.
+
+        """
+        try:
```
Comment (tomaskontrimas) [inline skyllh/core/dataset.py:1007]: change try ... except to if check

File: skyllh/core/dataset.py [hunk 7]
```diff
@@ -1088,6 +1120,21 @@ def get_aux_data_definition(self, name):
 
         return self._aux_data_definitions[name]
 
+    def remove_aux_data_definition(self, name):
```
Comment (tomaskontrimas) [inline skyllh/core/dataset.py:1123]: change try ... except to if check

File: skyllh/core/trialdata.py [hunk 8]
```diff
@@ -747,14 +749,6 @@ def get_data(self, name):
 
         if(name in self._source_data_fields_dict):
             data = self._source_data_fields_dict[name].values
-
```
Comment (martwo) [inline skyllh/core/trialdata.py:750]: This might break previous analyses that use scalar source data values.

**Pull request merged**